spring-security-training/src/main/java/com/example/springsecuritytraining/controller/AuthController.java

```diff
@@ -0,0 +1,128 @@
+package com.example.springsecuritytraining.controller;
+
+import com.example.springsecuritytraining.DTO.JwtResponseDTO;
+import com.example.springsecuritytraining.DTO.LoginRequestDTO;
+import com.example.springsecuritytraining.DTO.SignupRequestDTO;
+import com.example.springsecuritytraining.model.ERole;
+import com.example.springsecuritytraining.model.Role;
+import com.example.springsecuritytraining.model.User;
+import com.example.springsecuritytraining.repository.RoleRepository;
+import com.example.springsecuritytraining.repository.UserRepository;
+import com.example.springsecuritytraining.security.service.UserDetailsImpl;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.*;
+import com.example.springsecuritytraining.security.jwt.JwtUtils;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@CrossOrigin(origins = "*", maxAge = 3600)
+@RestController
+@RequestMapping("/")
```

Comment:
Use `/api` prefix. Reason for this is, usually the frontend and backend are deployed in the same origin.
Ex: google.com/login is used for fronend for the end users to see. google.com/api/auth/login is used for backend